lsコマンドを作る4を作成

sochi419 opened the pull request.

sochi419 committed:
Add ls4

sochi419 committed:
Changed method name and branching method

sochi419 commented:
以下2点を変更しました。
1. メソッドの分岐方法の変更
→rubocopにメソッドが長いと警告されたので、`get_type'と'get_permission'メソッド内の分岐内容を変更しました。

2. メソッドの名前の変更
→メソッド名が名詞だったので、動詞に変更しました。

確認よろしくお願いします🙏

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+
+    puts "total #{@total.sum}"
+
+    @file_details.each do |x|
```
おそらく次の行を短くしたいのでブロック変数も短い名前にしたのだと思いますが、せめて`f`とか`fd`とかにした方がまだ良いかなと思います。
理想的にはちゃんと`file_detail`という名前にして、表示のところの処理の形を変えてあげる方が良いとは思います。

sochi419 replied:
ブロック変数名を`f`に変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+    end
+  end
+
+  def get_total(file)
+    fs = File::Stat.new(file).blocks
+    @total << fs
+  end
```
`get`で始まるメソッドなのに、インスタンス変数にデータを格納するという副作用があるのはかなり不自然です。普通`get`で始まるには副作用がないことを期待しますし、何か欲しい値を返してくれることだけを期待します。

また、そもそも`@total`というインスタンス変数は存在意義がなさそうです。`file_detail`の中に入れられたらまた次のファイルの値で上書きされるだけなので、インスタンス変数というスコープの長い変数として保持される意味がありません。ですので、ここではインスタンス変数は使わずに`File::Stat.new(file).blocks`の値を戻り値として返すだけにしましょう。

他の`get_xxxx`メソッドも同様です。

sochi419 replied:
> getで始まるメソッドなのに、インスタンス変数にデータを格納するという副作用があるのはかなり不自然です。普通getで始まるには副作用がないことを期待しますし、何か欲しい値を返してくれることだけを期待します。

メソッド内のインスタンス変数にデータを収納する処理を削除しました。

> ここではインスタンス変数は使わずにFile::Stat.new(file).blocksの値を戻り値として返すだけにしましょう。

File::Stat.new(file).blocksを戻り値としました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+    end
+  end
+
+  def get_total(file)
```
`total`というより`blocks`ではないかという風に感じました。`total`、つまり「合計」なのは63行目で`sum`した後であって、この時点では`total`ではない気がします。

sochi419 replied:
get_blocksに変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+    puts "total #{@total.sum}"
+
+    @file_details.each do |x|
+      puts "#{x[0]}#{x[1]} #{x[2].rjust(@max_hardlink)} #{x[3].rjust(@max_user)} #{x[4].rjust(@max_group)} #{x[5].rjust(@max_filesize)} #{x[6]} #{x[7]}"
```
`x[0]`とか`x[1]`に何が入っているのかというのが分かりづらいですね。57行目と付き合わせながらでないと、何番目の要素に何が入っているのかわからないですよね。これでは他の人がコードを読んで理解したり、未来の自分がメンテしていくには結構厳しいと思います。

こういう時はハッシュを使いましょう。`x[0]`だと何が入っているか分かりませんが、`file_detail[:type]`なら多分ファイルタイプが入っているのだろうと推測できます。ちゃんと名前をつけてあげることは大切です。

sochi419 replied:
ハッシュを使って、何が入っているか理解しやすいようにしました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+
+  def adjust_space
+    i = 0
+    @hardlinks = []
+    @users = []
+    @groups = []
+    @filesizes = []
```
これらの変数はこの`adjust_space`メソッド内でしか使われないため、インスタンス変数である必要がありません。普通のローカル変数で十分です。（基本的に、変数のスコープはなるべく小さい（=寿命が短い)方が良いと考えてください。インスタンス変数のスコープは非常に大きいです。

sochi419 replied:
インスタンス変数から、ローカル変数に変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+      i += 1
+    end
+
+    @max_hardlink = @hardlinks.max_by(&:length).length
+    @max_user = @users.max_by(&:length).length
+    @max_group = @groups.max_by(&:length).length
+    @max_filesize = @filesizes.max_by(&:length).length
```
これらもインスタンス変数に入っていますが、後で一回使われるだけですし戻り値として返してあげるだけで良いと思います。

sochi419 replied:
calculate_spaceメソッドに、戻り値を返した処理に変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+    @groups = []
+    @filesizes = []
+
+    until i == @file_details.length
+      @hardlinks << @file_details[i][2]
+      @users << @file_details[i][3]
+      @groups << @file_details[i][4]
+      @filesizes << @file_details[i][5]
+      i += 1
+    end
```
`@file_details.each do |file_detail|`でループするのと同じだと思いますので、こちらの方が良いと思います。`until`や`while`を使うと自分で終了条件や配列の添字を指定する必要があるため、バグを混入する可能性が高くなります。例えば、`i`の値が想定と違っていて無限ループしてしまったり、配列の添字を間違えてデータアクセスしてしまう等です。`each`ではそういうことは起こらないので安全です。

sochi419 replied:
> untilやwhileを使うと自分で終了条件や配列の添字を指定する必要があるため、バグを混入する可能性が高くなります。

そうなんですね~
今後は`until`や`while`は使わずに、`each`を使うように心がけたいと思います！
`until`文を`each`文に書き換えました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+    @filesizes = []
+
+    until i == @file_details.length
+      @hardlinks << @file_details[i][2]
+      @users << @file_details[i][3]
+      @groups << @file_details[i][4]
+      @filesizes << @file_details[i][5]
```
ここでも`@file_details`（の中身の各要素）が配列であることの弊害が出ていますね。`@file_details[i][2]`に何が入っているかを推測するのは至難の業です。コードリーディングも大変です。これが`file_detail[:hardlink]`であればほぼ一目瞭然と言って良いでしょう。というわけでハッシュを使いましょう。

sochi419 replied:
ハッシュを使う書き方に変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+    @timestamp = File.mtime(file).strftime('%m %d %H:%M')
+  end
+
+  def adjust_space
```
この時点ではspaceをadjustしているわけではない（実際しているのは`rjust`するところ）ので、メソッド名はちょっと考え直した方が良いかなと思いました。

sochi419 replied:
確かに仰る通りですね。
adjust_spaceメソッドでは、空白の量を計算しているメソッドなので、
`間隔を計算する`という意味で、`calculate_space`に名前を変えました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,151 @@
+    @file_details = []
+    @total = []
+
+    @directories.each do |file|
+      get_total(file)
+      get_type(file)
+      get_permission(file)
+      get_hardlink(file)
+      get_user(file)
+      get_group(file)
+      get_size(file)
+      get_time(file)
+
+      file_detail = [@type, @permission_rwx, @hardlink, @user, @group, @filesize, @timestamp, file]
+      @file_details << file_detail
+    end
```
`each`ではなく`map`を使うことで、`@directories`から`@file_details`へのマッピング処理としてより簡潔に書けそうだなと思いました。こういうのがまさに`map`を使うべき場面だと思いますので、試してみてもらいたいです。

sochi419 replied:
mapを使った処理に変更しました。

summertree128 reviewed (commented):
レビューしました！
全体的にはよくできていると思いました。動作も問題なさそうです。

少し気になるのは、インスタンス変数が多い点です。インスタンス変数はローカル変数と比べてスコープが大きく、そのクラスのオブジェクトが生きている限り生存し続けます。
ローカル変数はメソッド内でのみ有効で、メソッドの実行が終わるとアクセスできなくなります。
基本的に変数のスコープが小さい方が良いと考えてもらいたいです。特にこの辺りはインスタンス変数である必要がないなというところにコメントしました。ご確認ください。

あとは配列の添字を使ってのデータアクセスが散見されますが、これは可読性が非常に低くメンテの難しいコードになってしまっています。こういう場合はハッシュを使いましょう。というコメントもしてあります。詳しくはコード内のレビューをご参照くださいー

sochi419 committed:
・I switch to the processing using the return value without using the instance variable
・I switch to the processing using the hash
・Not each, I switch to the processing using the map method
・Method name change

sochi419 commented:
レビューありがとうございました！
レビューをもとに、プログラムを修正しました！

主な変更点は以下です。
- インスタンス変数を使わず、メソッドの戻り値を使った処理に変更
- 配列名[0]の書き方が読みづらいため、ハッシュを使った表現に変更
- each構文の一部を、map構文に変更
- メソッド名を変更

確かに、ハッシュを使った方が、コードを読むときに読み手に理解しやすいと私も指摘いただいてから感じました。

今までの課題では、戻り値やハッシュは難しそうで使ってこず、苦手意識があったのですが、今回の修正で、メソッドに複数の戻り値を持たせられることや、ハッシュによる表記方法など新たな学びがあって良かったです！

確認よろしくお願いします🙏

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,134 @@
+    @directories.map! do |file|
+      total << get_blocks(file)
+      { type: get_type(file), permission: get_permission(file), hardlink: get_hardlink(file), user: get_user(file), group: get_group(file), size: get_size(file),
+        time: get_time(file), filename: file }
```
こういう感じで書いてあげた方がみやすいかもしれません。好みの問題でもありますが参考までに。

```suggestion
      { 
        type: get_type(file),
        permission: get_permission(file),
        hardlink: get_hardlink(file),
        user: get_user(file),
        group: get_group(file),
        size: get_size(file),
        time: get_time(file),
        filename: file
      }
```

sochi419 replied:
上記の記法に変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,134 @@
+  def run_option_l
+    total = []
+
+    @directories.map! do |file|
```
破壊的メソッドの使用はなるべく避けてください。特に慣れないうち…というかフィヨルドブートキャンプ卒業までくらいの間は原則的に使わないくらいの気持ちでいてもらいたいです。
今回だと`map`で書けると思うので`map`にしてくださいー。

参考：https://bootcamp.fjord.jp/pages/do-not-use-destructive-method

sochi419 replied:
> 破壊的メソッドの使用はなるべく避けてください。特に慣れないうち…というかフィヨルドブートキャンプ卒業までくらいの間は原則的に使わないくらいの気持ちでいてもらいたいです。

破壊的だと、バグとかの原因になったりするんですね...
今後は破壊的メソッドは控えるようにします👍

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,134 @@
+
+    puts "total #{total.sum}"
+
+    @directories.each do |f|
```
`@directries`なのであればブロックパラメータは`|directory|`か、もし一文字にするなら`d`がいいと思います

sochi419 replied:
hashなので、ブロック変数は|hash|にしました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,134 @@
+    puts "total #{total.sum}"
+
+    @directories.each do |f|
+      puts "#{f[:type]}#{f[:permission]} #{f[:hardlink].rjust(calculate_space[0])} #{f[:user].rjust(calculate_space[1])} #{f[:group].rjust(calculate_space[2])} #{f[:size].rjust(calculate_space[3])} #{f[:time]} #{f[:filename]}"
```
ここがRubocopに指摘されているところですね。長くなってしまっているので、文字列を組み立てる処理と出力する処理を分けて書いてみましょう。
```ruby
# 最終的に出力したい文字列を組み立てる処理
# 最終的に出力したい文字列を組み立てる処理
.....(数行かけて処理）

puts 最終的に出力したい文字列
```

こんな感じのイメージです。一つの要素につき一行書けばいいので、ブロックパラメータの名前が長くても気になりません。変に略さず`|directory|`とすることをお勧めします。

sochi419 replied:
確かに処理と出力を別々にすれば、文字列長さを指摘されなそうですね。
処理と出力を別々にしました👍

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,134 @@
+      filesizes << directory[:size]
+    end
+
+    [hardlinks.max_by(&:length).length, users.max_by(&:length).length, groups.max_by(&:length).length, filesizes.max_by(&:length).length]
```
これもハッシュにした方が良いですね。1番目にはhardlinkが、2番目にはusersが入っているみたいな考え方はだいぶ分かりづらいと思います。

sochi419 replied:
ハッシュに変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,134 @@
+  def get_permission(file)
+    permission = File::Stat.new(file).mode.to_s(8).chars
+    permission_integer = [permission[-3], permission[-2], permission[-1]]
+    permission_rwx = permission_integer.map do |y|
```
`y`ってなんでしたっけ？変数には意味のある名前をつけてもらいたいです。

sochi419 replied:
yは数字なので、ブロック変数は|number|としました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,134 @@
+  def sort_directories
+    sorted_directories = divide_directories.transpose
+    max_length = @directories.max_by(&:length).length
+    sorted_directories.each do |x|
+      x.each do |y|
```
この辺の`x`とか`y`もちゃんと意味のある名前にしてもらいたいです。
なかなかいい名前を考えるのが難しいところではありますが…

sochi419 replied:
ブロック変数はx,yのように、特に意味のないただの変数として、定義していました。
今後はx,yなど意味の無い変数名ではなく、意味ある変数名にします。

x→|array|
y→|directory|
に変数名を変更しました。

sochi419 committed:
・I change destructive processing
・I add total method
・I change the output and the handling of run_option_l method separately
・I change a block variable name

sochi419 commented:
レビュー有難うございます！

下記の点を修正しました↓
・破壊的処理を変更
・run_option_lメソッドの出力と処理を別々に変更
・ブロック変数名を変更

また、上記の変更を加えたところ、run_option_lメソッドが長いとrubocopに指摘されました。
そのため、run_option_lメソッドを切り分けて、新たに以下のメソッドを追加しました。
・convert_file_to_hashメソッドを追加
・totalメソッドを追加

以上、確認よろしくお願いします🙏

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -32,28 +32,52 @@ def divide_directories
-    sorted_directories.each do |x|
-      x.each do |y|
-        print format("%-#{max_length + 1}s", y)
+    sorted_directories.each do |array|
```
`array`はただのデータ型であって、そこに何が入っているかという本質的な情報ではありません。
基本的に、変数名にはデータ型に関する情報を入れない方が良いです。（まれに入れたくなる時はありますが、大抵はない方が良いです）
`each`などの繰り返し処理の場合は、繰り返す対象のコレクション（今回で言うと`sorted_directories`）の単数系にするのが定石です。つまり`sorted_directory`です。

sochi419 replied:
ブロック変数名を`sort_directory`に変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -32,28 +32,52 @@ def divide_directories
-        time: get_time(file), filename: file }
+    puts "total #{total}"
+
+    convert_file_to_hash.each do |hash|
```
hashはただのデータ型であって、そこに何が入っているかという本質的な情報ではありません。
いま`hash`に入っているのはなんですか？人に説明するとしたら何と説明しますか？それを英語にしたものを名前にしてもらいたいです。

sochi419 replied:
convert_file_to_hashは削除しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -114,14 +138,21 @@ def calculate_space
 
-    [hardlinks.max_by(&:length).length, users.max_by(&:length).length, groups.max_by(&:length).length, filesizes.max_by(&:length).length]
+    max_lengths = {
+      hardlink_max: hardlinks.max_by(&:length).length,
+      user_max: users.max_by(&:length).length,
+      group_max: groups.max_by(&:length).length,
+      filesize_max: filesizes.max_by(&:length).length
```
変数名で`max_length`が入っていることはわかるので、キー名にまで`max`をつけなくてもいいかなとおもいました。

sochi419 replied:
キー名のmaxは消去しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,165 @@
+  end
+
+  def get_blocks(file)
+    File::Stat.new(file).blocks
```
`File::Stat.new(file)`または`File.stat(file)`がいろんなメソッドの中で実行されていて無駄なので、一つのファイルに対して一度だけ実行されるようにしてください。

sochi419 replied:
````
def get_file_info(file)
    File.stat(file)
end
````
`get_file_info(file) `メソッドを作成して、戻り値を`File.stat(file)`としました。
このメソッドを何度も使用する処理に変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,165 @@
+    groups = []
+    filesizes = []
+
+    convert_file_to_hash.each do |hash|
```
`convert_file_to_hash`が複数回実行されていてかなり無駄が大きいので、一度だけ実行されるようにしてください。

sochi419 replied:
`convert_file_to_hash`メソッドは削除しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,165 @@
+  end
+
+  def run_option_l
+    puts "total #{total}"
```
個人的な意見ですが、文字列中の変数展開でメソッドを呼び出すと言うのは結構分かりづらい処理だと思います。（この`total`っていう変数、どこで定義されてるんだ？と思ってしまいました）
ローカル変数に格納してから変数展開した方が良いと思います。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,165 @@
+    end
+  end
+
+  def total
```
このメソッドだけ動詞はじまりになっていないのは何か理由があるのでしょうか？

sochi419 replied:
ミスで名詞にしてしまいました。
`calculate_blocks(files)`としました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,165 @@
+    sorted_directories = divide_directories.transpose
+    max_length = @directories.max_by(&:length).length
+    sorted_directories.each do |array|
+      array.each do |directory|
+        print format("%-#{max_length + 1}s", directory)
+      end
+      puts
```
`sort`というメソッド名なのに出力まで行ってしまうのはだいぶ違和感がありますね。出力部分は別のメソッドに切り出した方が良いと思います。

sochi419 replied:
ディレクトリの配置に関するメソッド`sort_directories(files)`と、出力のメソッド`output(files)`に切り分けました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,165 @@
+
+  def divide_directories
+    if (@directories.size % COLUMN).zero?
+      divide_directories = @directories.each_slice(@directories.size / COLUMN).to_a
```
変数名としては`divided_directories`の方が良さそうですね。

sochi419 replied:
変数名`divided_directories`に変更しました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,165 @@
+      filesize_max: filesizes.max_by(&:length).length
+    }
+
+    [max_lengths[:hardlink_max], max_lengths[:user_max], max_lengths[:group_max], max_lengths[:filesize_max]]
```
せっかくハッシュにしたのにここで配列にしてしまったら勿体無いです。。結局これだと最終的に添字でアクセスすることになってしまっています。

このメソッドを利用する側が、配列の添字ではなくハッシュのキーを使ってmax lengthを取り出せるようにしてもらいたいです。

sochi419 replied:
配列のところを下記の表記に変更しました。
````
  {
      hardlink: hardlinks.max_by(&:length).length,
      user: users.max_by(&:length).length,
      group: groups.max_by(&:length).length,
      filesize: filesizes.max_by(&:length).length
  }
````

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,165 @@
+    convert_file_to_hash.each do |hash|
+      type = hash[:type]
+      permission = hash[:permission]
+      hardlink = hash[:hardlink].rjust(calculate_space[0])
+      user = hash[:user].rjust(calculate_space[1])
+      group = hash[:group].rjust(calculate_space[2])
+      size = hash[:size].rjust(calculate_space[3])
```
`calculate_space`をファイル数 * 4回実行していますが、実際には一度だけ実行すれば十分なはずです。結果は毎回同じなので。というわけで`calculate_space`が一度しか実行されないように修正をお願いします。

sochi419 replied:
何度も`calculate_max`メソッドが実行されない様に、
' max_length = calculate_max'として、`run_option_l(file)`メソッドの初めに`calculate_max`の戻り値を変数で定義しました。
定義した変数`max_length`を何度も実行する処理に変更しました。

summertree128 reviewed (commented):
再度レビューしました！

またプラクティスの方にも包括的なコメントをしてありますのでご確認お願いしますー。

sochi419 committed:
The changes are as follows.
1. I switch to the processing using the main method
2. Variable name
3. Deletion of the useless processing
4. It is cut the method into pieces

sochi419 committed:
The changes are as follows.
1. I switch to the processing using the main method
2. Variable name
3. Deletion of the useless processing
4. It is cut the method into pieces

sochi419 commented:
長らくレビューの返信が遅くなり申し訳ありません....!
修正しました！
主な変更点は以下になります。

1. mainメソッドを用いた書き方に変更。
→前回まではclassを定義して、lsコマンドで表示されるディレクトリ群を@directoriesと表現していた。
メソッドに引数を渡す際、グローバル変数は好ましくないので、mainメソッドに変更しファイル名をfilesとしました。
2. 引数を受け取るメソッドに変更
3. 変数名の修正
4. 無駄な処理の削除/修正
5. メソッドの切り分け

確認よろしくお願いします🙏

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -3,163 +3,150 @@
-        print format("%-#{max_length + 1}s", directory)
-      end
-      puts
+def divided_directories(files)
```
メソッド名は動詞（の現在形）から始めよ、ということで`divide_directories` の方がよさそうですね。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,152 @@
+          end
+
+  if opt['l']
+    calculate_blocks(files)
```
`calculate`とあるので、このメソッドはブロック数を計算するのかな？という風に読めますが、戻り値が使われていないので、「計算結果はどこに行ったのかな？」というのが第一印象でした。
実際にメソッドの中身を読んでみると、計算したあと標準出力への出力までやっていますね。ちょっと名前と実態が乖離してしまっているように感じます。”ブロック数を出力する”みたいな感じのメソッド名にしてしまえばいいかなと思いました。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,152 @@
+    calculate_blocks(files)
+
+    files.each do |file|
+      run_option_l(file)
```
`calculate_blocks`, `run_option_l`, `output`という三つのメソッド、これらは全て`main`メソッドから呼ばれ、標準出力にlsの結果を出力するという点で非常に似ていますね。抽象度が同じレベルだと言ってもいいと思います。
こういう似たようなレベルのメソッド達は似たような命名をしてあげると非常にわかりやすくなります。例えば全部同じ動詞で始めるとかでしょうか。ちょっと検討してみてもらいたいです！

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,152 @@
+  puts "#{type}#{permission} #{hardlink} #{user} #{group} #{size} #{time} #{filename}"
+end
+
+def calculate_max
```
maxとは？と思わなくもないので、`calculate_max_lengths`とかにしておいた方がわかりやすいかもしれませんね。

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,152 @@
+  groups = []
+  filesizes = []
+
+  Dir.glob('*').each do |file|
```
`Dir.glob('*')`はもう実行しているはずなのでここでもう一度実行するのは無駄ですね。ファイルの一覧を引数でもらいましょう。

sochi419 committed:
I changed the following points.
・I change the naming of the method of it
・Dir.glob(*) changes only once to the processing carried out

sochi419 commented:
レビューありがとうございます！
修正しました、
主な変更点は以下になります。

1. メソッド名の変更
・main下の似たような（抽象度が同じ程度の）メソッド名は、統一性のある名前に変更した。
・メソッド名をより具体的な名前に変更しました。
2. Dir.glob(*)が何度も実行されている処理を変更
メソッドに引数を用意することで、Dir.glob(*)が何度も実行されない処理に変更しました。

以上、確認よろしくお願いします🙏

summertree128 commented on 05.ls/ls4.rb:
```diff
@@ -0,0 +1,154 @@
+
+main
+
+
```
空行が一行余計な気がしますー

summertree128 reviewed (approved):
いいんじゃないでしょうか！OKです！

sochi419 committed:
remove blank line

sochi419 committed:
remove blank line
